iroh-resolver/src/resolver.rs

```diff
@@ -225,22 +229,33 @@ pub enum UnixfsType {
 }
 
 pub enum OutPrettyReader<T: ContentLoader> {
-    DagPb(usize, Bytes),
+    DagPb(usize, Bytes, iroh_metrics::gateway::Metrics, Instant),
     Unixfs(UnixfsReader<T>),
-    DagCbor(usize, Bytes),
-    DagJson(usize, Bytes),
-    Raw(usize, Bytes),
+    DagCbor(usize, Bytes, iroh_metrics::gateway::Metrics, Instant),
+    DagJson(usize, Bytes, iroh_metrics::gateway::Metrics, Instant),
+    Raw(usize, Bytes, iroh_metrics::gateway::Metrics, Instant),
 }
 
 impl Out {
-    pub fn pretty<T: ContentLoader>(self, loader: T) -> OutPrettyReader<T> {
+    pub fn pretty<T: ContentLoader>(
+        self,
+        loader: T,
+        metrics: iroh_metrics::gateway::Metrics,
+        start_time: Instant,
```

Comment:
This is really messy having to pass the time manually around :/ can't we store that in the `Metrics` struct maybe?